src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,205 @@
+package com.faforever.client.teammatchmaking;
+
+import com.faforever.client.domain.api.MapVersion;
+import com.faforever.client.domain.api.MatchmakerQueueMapPool;
+import com.faforever.client.domain.server.PlayerInfo;
+import com.faforever.client.fx.FxApplicationThreadExecutor;
+import com.faforever.client.fx.JavaFxUtil;
+import com.faforever.client.player.LeaderboardRating;
+import com.faforever.client.player.PlayerService;
+import com.faforever.client.theme.UiService;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.layout.FlowPane;
+import javafx.scene.layout.Pane;
+
+import com.faforever.client.domain.server.MatchmakerQueueInfo;
+import com.faforever.client.fx.NodeController;
+import com.faforever.client.map.MapService;
+import javafx.scene.layout.VBox;
+import javafx.stage.Screen;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@RequiredArgsConstructor
+
+public class TeamMatchmakingMapListController extends NodeController<Pane> {
+
+  private static final int TILE_SIZE = 125;
+  private static final int PADDING = 20;
+
+  private final MapService mapService;
+  private final UiService uiService;
+  private final PlayerService playerService;
+  private final FxApplicationThreadExecutor fxApplicationThreadExecutor;
+
+  public Pane root;
+  public FlowPane tilesContainer;
+  public ScrollPane scrollContainer;
+  public VBox loadingPane;
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets;
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBracketsWithDuplicates;
+  private Integer playerBracketIndex = null;
+  @Setter
+  private double maxWidth = 0;
+  @Setter
+  private double maxHeight = 0;
```

Comment:
These should be properties rather than just primitives so they can be responsive.